core/elements/ons-tab.spec.es6

```diff
@@ -0,0 +1,240 @@
+
+      expect(element.children[0]).to.be.ok;
+      expect(element.children[1]).to.be.ok;
+      expect(element.children[2]).not.to.be.ok;
+
+      expect(element.children[0].nodeName).to.equal('INPUT');
+      expect(element.children[0].type).to.equal('radio');
+      expect(element.children[0].style.display).to.equal('none');
+
+      expect(element.children[1].nodeName).to.equal('BUTTON');
+      expect(element.children[1].classList.contains('tab-bar__button')).to.be.true;
+      expect(element.children[1].classList.contains('tab-bar-inner')).to.be.true;
+    });
+  });
+
+  describe('parent', () => {
+    it('should be a \'ons-tabbar\' element', () => {
+      expect(() => element._ensureElementPosition()).to.throw('This ons-tab element is must be child of ons-tabbar element.');
+
+      var parent = new OnsTabbarElement();
+      parent.appendChild(element);
+      expect(() => element._ensureElementPosition()).not.to.throw('This ons-tab element is must be child of ons-tabbar element.');
+    });
+  });
+
+  describe('_hasDefaultTemplate property', () => {
+    it('is, by defaut, true', () => {
+      expect(element._hasDefaultTemplate).to.be.true;
+    });
+
+  it('is false when one of the tab\'s children is a ELEMENT_NODE', () => {
+      let tabbar = new OnsTabbarElement();
+      document.body.appendChild(tabbar);
+      element = ons._util.createElement(`
+        <ons-tab active="true">
+          <div></div>
+        </ons-tab>
+      `);
+      tabbar.appendChild(element);
+      expect(element._hasDefaultTemplate).not.to.be.true;
+    });
+  });
+
+  describe('#_updateDefaultTemplate()', () => {
+    it('will return if there is not a default template', () => {
+      var spy = chai.spy.on(element, 'getAttribute');
+      element._hasDefaultTemplate = false;
+      element._updateDefaultTemplate();
+      expect(spy).not.to.have.been.called;
+    });
+  });
+
+  describe('#_loadPage()', () => {
+    it('returns the current tab _pageElement, if the tab has persistent attribute and a _pageElement', (done) => {
+      element = ons._util.createElement(`
+        <ons-tab persistent>
+        </ons-tab>
+      `);
+
+      let myFunction = (value) => {
+        expect(value).to.equal(element._pageElement);
+        done();
+      };
+      element._pageElement = true;
+      element._loadPageElement(myFunction);
+    });
+
+    it('sets the tab _pageElement as null, if the tab doesn\'t has a persistent attribute', (done) => {
+      let myFunction = () => {
+        expect(element._pageElement).to.be.null;
+        done();
+      };
+      element._loadPageElement(myFunction);
+    });
+  });
+
+  describe('#setActive()', () => {
+    it('will set the tab as active', () => {
+      let tabbar = new OnsTabbarElement();
+
+      document.body.appendChild(tabbar);
+      document.body.innerHTML += '<ons-template id="page1"><ons-page></ons-page></ons-template><ons-template id="page2"><ons-page></ons-page></ons-template>';
```

Comment:
If you add elements like this you can't remove them easily. Please create with `ons._util.createElement` and then remove them after the test is finished.